cc3d/core/PyCoreSpecs.py

```diff
@@ -3116,12 +3157,19 @@ class ContactPlugin(_PyCorePluginSpecs):
     name = "contact"
     registered_name = "Contact"
 
-    check_dict = {"neighbor_order": (lambda x: x < 1, "Neighbor order must be positive")}
+    check_dict = {"neighbor_order": (lambda x: x is not None and x < 1, "Neighbor order must be positive"),
+                  "depth": (lambda x: x is not None and x < 1, "Depth must be positive")}
```

Comment:
I am not sure about the `check_dict` approach. By setting `self.check_dict` rather than passing in check_dict to `self.check_inputs()`, we increase the potential for race conditions (if any of this is multi-threaded). It would be simple & concise to use if-else statements as opposed to this method.

Reply:
Since `check_dict` is a class attribute, race conditions shouldn't be a problem. Did you have a specific scenario in mind where a race condition might occur?

Reply:
Ah, you are right. At first, I read it as static.